### PR TITLE
[Agency Dashboard] Share Screen scaffolding

### PR DESCRIPTION
**Pull request by terryttsai**
## Description of the change
Screen Shot 2022-12-14 at 3 25 40 PM
Screen Shot 2022-12-14 at 3 26 35 PM
Screen Shot 2022-12-14 at 3 25 59 PM

Share screen (minus the option to share w the current filters, that's TBD in this PR: https://github.com/Recidiviz/justice-counts/issues/219)

Added a toast that says "Copied!" when you click the Copy URL button.

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #XXXX

### Development

**This box MUST be checked by the submitter prior to merging**:
- [ ] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: agency-dashboard/src/DashboardModals/DashboardModal.styles.tsx [hunk 1]
```diff
@@ -102,7 +101,66 @@ export const LearnMoreModalParagraph = styled.div`
...
+  color: ${palette.highlight.grey10};
+  border-radius: 2px;
+  height: 48px;
+  text-overflow: ellipsis;
+  padding: 14px 16px;
+  flex: 1;
+  ${typography.sizeCSS.normal}
+`;
+
+export const ShareBarButton = styled.div`
```
Comment (terryttsai) [inline agency-dashboard/src/DashboardModals/DashboardModal.styles.tsx:130]: Could put these text boxes, buttons, checkboxes in a more generic location if they're going to be reused designs.

File: agency-dashboard/src/DashboardModals/ShareModal.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          <CloseIcon />
+        </ModalCloseButton>
+        <ModalInnerContainer>
+          <ModalTitle>Share</ModalTitle>
+          <ModalParagraph>
+            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nec vel vel
+            nunc lacus diam varius varius enim risus.
+          </ModalParagraph>
+          <ShareBarContainer>
+            <ShareBarURLBox readOnly value={window.location.href} />
```
Comment (mxosman) [inline agency-dashboard/src/DashboardModals/ShareModal.tsx:51]: Nice! Haven't seen the `readOnly` prop before. Good to know!

File: common/components/Toast/Toast.ts [hunk 3]
```diff
@@ -30,7 +30,8 @@ export const showToast = (
   check = false,
   color: ToastColor = "blue",
   timeout = 2500,
-  preventOverride?: boolean
+  preventOverride = false,
+  positionNextToIcon = true // in the Publisher app, there is an icon with 65px width at the top left corner of the screen
```
Comment (mxosman) [inline common/components/Toast/Toast.ts:34]: Nice!

**Comment by mxosman** [comment]
Looks pretty straightforward to me, Terry! This is awesome - I'll play around with this some more tomorrow morning before approving!

**Review by lilidworkin — COMMENTED** [review]
Nice @terryttsai ! Agreed that this looks straightforward, though I'll let @mxosman accept.

If you haven't already, would be good to test that you can indeed actually "paste" the copied text into a different application, and then paste it back into the search bar, and load the same page back again.

**Review by mxosman — APPROVED** [review]
Works beautifully!

**Comment by terryttsai** [comment]
Thanks all! Sorry I didn't get a playtesting link this time around, I ran into a few errors but I got it working now.

File: agency-dashboard/src/DashboardModals/DashboardModal.styles.tsx (hunk 1 above)
Comment (terryttsai) [reply agency-dashboard/src/DashboardModals/DashboardModal.styles.tsx]: @mxosman moved them to a generic primitives location, let me know what you think!

Comment (mxosman) [reply agency-dashboard/src/DashboardModals/DashboardModal.styles.tsx]: LOVE that, Terry! ✍️ 🗒️ Making a note for myself to do something similar in publisher!

**Pull request merged**